src/plugins/intel_cpu/src/nodes/eltwise.cpp

```diff
@@ -1909,7 +2050,23 @@ void Eltwise::getSupportedDescriptors() {
 }
 
 void Eltwise::initSupportedPrimitiveDescriptors() {
-    std::vector<Precision> supportedPrecisions = {
+    const auto isBitwise = [](const Algorithm& algorithm) {
+        return one_of(
+            algorithm,
+            Algorithm::EltwiseBitwiseAnd,
+            Algorithm::EltwiseBitwiseNot,
+            Algorithm::EltwiseBitwiseOr,
+            Algorithm::EltwiseBitwiseXor);
+    };
+
+    std::vector<Precision> supportedPrecisions = isBitwise(algorithm) ?
+        std::vector<Precision> {
+            Precision::U8,
+            Precision::I8,
+            Precision::U16,
+            Precision::I16,
+            Precision::I32
```

Comment:
Is boolean datatype supported? While testing torch frontend from https://github.com/openvinotoolkit/openvino/pull/20339 I see that there are some issues with BitwiseNot with boolean input

Reply:
it's supported, `boolean` type is converted into `u8` in CPU.

Reply:
Thanks, I guess that this might explain possible problem with BitwiseNot, True converted to u8 would be `00000001`, and after bitwise not operation it would flip all bits, resulting in 254 that would probably be converted back to True, resulting in BitwiseNot never returning values with False